tests/v2/validation/charts/backup_restore/backup_restore.go

```diff
@@ -0,0 +1,332 @@
+	createdSecret, err := steveClient.SteveType(secrets.SecretSteveType).Create(secretTemplate)
+
+	return createdSecret.Name, err
+}
+
+func createRancherResources(client *rancher.Client, clusterID string, context string) ([]*management.User, []*management.Project, []*management.RoleTemplate, error) {
+	userList := []*management.User{}
+	projList := []*management.Project{}
+	roleList := []*management.RoleTemplate{}
+
+	for i := 0; i < resourceCount; i++ {
+		u, err := users.CreateUserWithRole(client, users.UserConfig(), "user")
+		if err != nil {
+			return userList, projList, roleList, err
+		}
+		userList = append(userList, u)
+
+		p, _, err := projects.CreateProjectAndNamespace(client, clusterID)
+		if err != nil {
+			return userList, projList, roleList, err
+		}
+		projList = append(projList, p)
+
+		rt, err := client.Management.RoleTemplate.Create(
+			&management.RoleTemplate{
+				Context: context,
+				Name:    roleName,
+				Rules:   rules,
+			})
+		if err != nil {
+			return userList, projList, roleList, err
+		}
+		roleList = append(roleList, rt)
+	}
+
+	return userList, projList, roleList, nil
+}
+
+func createBackup(client *rancher.Client, bucket string) (*v1.SteveAPIObject, string, error) {
+	backup := setBackupObject()
+	backupTemplate := bv1.NewBackup("", backupName, backup)
+	completedBackup, err := client.Steve.SteveType(backupSteveType).Create(backupTemplate)
+	if err != nil {
+		return nil, "", err
+	}
+
+	charts.VerifyBackupCompleted(client, backupSteveType, completedBackup)
+
+	backupObj, err := client.Steve.SteveType(backupSteveType).ByID(backupName)
+	if err != nil {
+		return nil, "", err
+	}
+
+	backupK8Obj := &bv1.Backup{}
+	err = v1.ConvertToK8sType(backupObj.JSONResp, backupK8Obj)
+	if err != nil {
+		return nil, "", err
+	}
+
+	backupFileName := backupK8Obj.Status.Filename
+
+	client.Session.RegisterCleanupFunc(func() error {
+		err = deleteAWSObject(bucket, backupFileName)
+		if err != nil {
+			return err
+		}
+
+		_, err = checkAWSS3Object(bucket, backupFileName)
+		if err != nil {
+			s3Error := err.Error()
+			errorText := strings.Contains(s3Error, "404")
+			if !errorText {
+				return err
+			}
+		}
+
+		return nil
+	})
+
+	return completedBackup, backupFileName, err
+}
+
+func createDownstreamCluster(client *rancher.Client, clusterType string) (*management.Cluster, *v1.SteveAPIObject, *clusters.ClusterConfig, error) {
+	provisioningConfig := new(provisioninginput.Config)
+	config.LoadConfig(provisioninginput.ConfigurationFileKey, provisioningConfig)
+	nodeProviders := provisioningConfig.NodeProviders[0]
+	externalNodeProvider := provisioning.ExternalNodeProviderSetup(nodeProviders)
+	testClusterConfig := clusters.ConvertConfigToClusterConfig(provisioningConfig)
+	testClusterConfig.CNI = provisioningConfig.CNIs[0]
+
+	var clusterObject *management.Cluster
+	var steveObject *v1.SteveAPIObject
+	var err error
+
+	switch clusterType {
+	case "RKE1":
```

Comment:
why not just have 2 functions, one to createRKE1dscluster and another for createRKE2dscluster? It would be more clean since there are only 2 use cases.